components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/dao/RequestObjectDAOImpl.java

```diff
@@ -0,0 +1,362 @@
+/*
+ *
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ * /
+ */
+
+package org.wso2.carbon.identity.openidconnect.dao;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationManagementUtil;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.oauth.IdentityOAuthAdminException;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dao.AuthorizationCodeDAOImpl;
+
+import org.wso2.carbon.identity.oauth2.dao.OAuthTokenPersistenceFactory;
+import org.wso2.carbon.identity.openidconnect.OIDCConstants;
+import org.wso2.carbon.identity.openidconnect.model.RequestedClaim;
+import org.wso2.carbon.utils.DBUtils;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.oauth.OAuthUtil.handleError;
+
+/**
+ * This class handles all the DAO layer activities which are related to OIDC request object.
+ */
+public class RequestObjectDAOImpl implements RequestObjectDAO {
+
+    private static final String ID = "ID";
+    private final Log log = LogFactory.getLog(AuthorizationCodeDAOImpl.class);
+
+    /**
+     * Store request object related data into related db tables.
+     *
+     * @param consumerKey    consumer key
+     * @param sessionDataKey session data key
+     * @param claims         request object claims
+     * @throws IdentityOAuth2Exception
+     */
+    @Override
+    public void insertRequestObjectData(String consumerKey, String sessionDataKey, List<List<RequestedClaim>> claims)
+            throws IdentityOAuth2Exception {
+
+        int requestObjectId = -1;
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+        String sqlStmt = SQLQueries.STORE_IDN_OIDC_REQ_OBJECT_REFERENCE;
+        Connection connection = null;
+        try {
```

Comment:
shall we use try with resource for connections, result sets and prepared statements?